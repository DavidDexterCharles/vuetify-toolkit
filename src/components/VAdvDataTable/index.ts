import VAdvDataTable from './VAdvDataTable'

export { VAdvDataTable }
export default VAdvDataTable
