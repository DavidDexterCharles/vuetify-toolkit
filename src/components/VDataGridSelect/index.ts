import VDataGridSelect from './VDataGridSelect'

export { VDataGridSelect }
export default VDataGridSelect
