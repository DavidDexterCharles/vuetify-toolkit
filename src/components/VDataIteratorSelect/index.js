import VDataIteratorSelect from './VDataIteratorSelect'

export { VDataIteratorSelect }
export default VDataIteratorSelect
