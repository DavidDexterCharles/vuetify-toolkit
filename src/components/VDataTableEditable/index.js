import VDataTableEditable from './VDataTableEditable'

export { VDataTableEditable }
export default VDataTableEditable
