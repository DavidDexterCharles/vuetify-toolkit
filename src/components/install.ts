import { VueConstructor, Component } from 'vue'

export interface VuetifyToolkitUseOptions {
  components?: Record<string, Component>
}

export function install (Vue: VueConstructor, args: VuetifyToolkitUseOptions) {
  if ((install as any).installed) return
  (install as any).installed = true
  const components = args.components || {}

  for (const key in components) {
    const component = components[key]
    if (component) {
      Vue.component(key, component as typeof Vue)
    }
  }
  // Used to avoid multiple mixins being setup
  // when in dev mode and hot module reload
  // https://github.com/vuejs/vue/issues/5089#issuecomment-284260111
  if ((Vue as any).$_vuetify_toolkit_installed) return
  (Vue as any).$_vuetify_toolkit_installed = true
}
