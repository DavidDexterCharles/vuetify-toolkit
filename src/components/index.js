import VTreeSelect from './VTreeSelect'
import VCascader from './VCascader'
import VDataGridSelect from './VDataGridSelect'
import VMdView from './VMdView'

import Vuetify from 'vuetify/lib'

const install = Vuetify.install

const components = {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}

Vuetify.install = (Vue, args) => {
  install.call(Vuetify, Vue, {
    components
  })
}

export default {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}

export {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}
