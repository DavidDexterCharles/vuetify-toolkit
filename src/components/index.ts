import VTreeSelect from './VTreeSelect'
import VCascader from './VCascader'
import VDataGridSelect from './VDataGridSelect'
import VDateTimeSelect from './VDateTimeSelect'
import VMdView from './VMdView'
import VAdvDataTable from './VAdvDataTable'

export default {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VDateTimeSelect,
  VMdView,
  VAdvDataTable
}

export {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VDateTimeSelect,
  VMdView,
  VAdvDataTable
}
