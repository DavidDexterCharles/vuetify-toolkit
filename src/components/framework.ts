import Vue from 'vue'
import { install } from './install'

export default class VuetifyToolkit {
  static install = install

  static installed = false

  public installed: string[] = []

  // Called on the new vuetify-toolkit instance
  // bootstrap in install beforeCreate
  // Exposes ssrContext if available
  init (root: Vue, ssrContext?: object) {}
}
