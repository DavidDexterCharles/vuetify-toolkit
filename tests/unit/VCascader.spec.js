// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify/lib'
// Utilities
import {
  mount
} from '@vue/test-utils'
// component to be tested
import { VCascader } from '@/components'

Vue.use(Vuetify)

const vuetify = new Vuetify({
  icons: {
    iconfont: 'mdi'
  }
})

const singleRootTwoChildren = [
  { id: 0, name: 'Root', children: [{ id: 1, name: 'Child' }, { id: 2, name: 'Child 2' }] }
]

describe('VCascader.js', () => {
  it('renders', () => {
    const wrapper = mount(VCascader, {
      vuetify,
      propsData: {
        items: singleRootTwoChildren
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
