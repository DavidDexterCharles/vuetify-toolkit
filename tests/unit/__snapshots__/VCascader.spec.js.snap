// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCascader.js renders 1`] = `
<div class="v-input theme--light v-text-field v-select v-autocomplete">
  <div class="v-input__control">
    <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-1" class="v-input__slot">
      <div class="v-select__slot">
        <div class="v-select__selections"><input id="input-1" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden">
      </div>
      <div class="v-menu">
        <div class="v-menu__content theme--light v-autocomplete__content" style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
          <div role="listbox" tabindex="-1" class="v-list v-select-list v-sheet v-sheet--tile theme--light theme--light">
            <div tabindex="0" aria-selected="false" id="list-item-6-0" role="option" class="v-list-item v-list-item--link theme--light">
              <div class="v-list-item__content">
                <div class="v-list-item__title">[object Object]</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
    </div>
  </div>
</div>
`;
